pkg/machine/shim/host.go

```diff
@@ -467,6 +480,91 @@ func Start(mc *vmconfigs.MachineConfig, mp vmconfigs.VMProvider, dirs *machineDe
 	return nil
 }
 
+func Set(mc *vmconfigs.MachineConfig, mp vmconfigs.VMProvider, opts machineDefine.SetOptions) error {
+	mc.Lock()
+	defer mc.Unlock()
+
+	if err := mp.SetProviderAttrs(mc, opts); err != nil {
+		return err
+	}
+
+	// Update the configuration file last if everything earlier worked
+	return mc.Write()
+}
+
+func Remove(mc *vmconfigs.MachineConfig, mp vmconfigs.VMProvider, dirs *machineDefine.MachineDirs, opts machine.RemoveOptions) error {
+	mc.Lock()
+	defer mc.Unlock()
+
+	state, err := mp.State(mc, false)
+	if err != nil {
+		return err
+	}
+
+	if state == machineDefine.Running {
+		if !opts.Force {
+			return &machineDefine.ErrVMRunningCannotDestroyed{Name: mc.Name}
+		}
+	}
+
+	rmFiles, genericRm, err := mc.Remove(opts.SaveIgnition, opts.SaveImage)
+	if err != nil {
+		return err
+	}
+
+	providerFiles, providerRm, err := mp.Remove(mc)
+	if err != nil {
+		return err
+	}
+
+	// Add provider specific files to the list
+	rmFiles = append(rmFiles, providerFiles...)
+
+	// Important!
+	// Nothing can be removed at this point.  The user can still opt out below
+	//
+
+	if !opts.Force {
+		// Warn user
+		confirmationMessage(rmFiles)
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Print("Are you sure you want to continue? [y/N] ")
+		answer, err := reader.ReadString('\n')
+		if err != nil {
+			return err
+		}
+		if strings.ToLower(answer)[0] != 'y' {
+			return nil
+		}
```

Comment:
So this function is a bit special because we prompt the user here which means we can block for ever if the user does not respond.
I think we should unlock again before ask the user and then lock again after they did but this means we need to reald the file after we lock again.

Alternately keeping it locked could also serve as protection so that nobody modifies thing in the meantime so maybe this fine right now as is.

Reply:
I think I'd prefer to keep it locked. What happens if multiple processes try to RM a machine where in the non locked portion? I feel like something would go weird there. Worst comes to worse, we can edit it later.

Reply:
yeah I think keeping it locked is the easiest for us to deal with and to keep it consistent for now

Reply:
could spin off a timer that autocancels if long-term lock holding is a concern ... but, machine is a really a single user thing.  if they forget to ack the prompt and start issuing other commands, hopefully they can put 2 and 2 together ?

Reply:
I think a time would be a good idea to address in the follow-up pr.